Mage/src/main/java/mage/abilities/common/PutCounterOnCreatureTriggeredAbility.java

```diff
@@ -0,0 +1,90 @@
+package mage.abilities.common;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.constants.Zone;
+import mage.counters.Counter;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ * "Whenever you put one or more counters on a creature " triggered ability
+ *
+ * @author PurpleCrowbar
+ */
+public class PutCounterOnCreatureTriggeredAbility extends TriggeredAbilityImpl {
+
+    private final Counter counterType; // when null, any counter type is accepted
+    protected FilterPermanent filter;
+    private final boolean setTargetPointer;
+
+    public PutCounterOnCreatureTriggeredAbility(Effect effect) {
+        this(effect, false);
+    }
+
+    public PutCounterOnCreatureTriggeredAbility(Effect effect, boolean optional) {
+        this(effect, null, new FilterCreaturePermanent(), optional, false);
+    }
+
+    public PutCounterOnCreatureTriggeredAbility(Effect effect, Counter counter, boolean optional) {
+        this(effect, counter, new FilterCreaturePermanent(), optional, false);
+    }
+
+    public PutCounterOnCreatureTriggeredAbility(Effect effect, FilterPermanent filter, boolean optional) {
+        this(effect, null, filter, optional, false);
+    }
+
+    public PutCounterOnCreatureTriggeredAbility(Effect effect, Counter counter, FilterPermanent filter, boolean optional, boolean setTargetPointer) {
```

Comment:
is the `optional` parameter needed? All of the abilities which use these are non-optional

Reply:
I noticed that too and was going to get rid of the optional parameter, but decided to keep it in for any future uses of the ability which may be optional

Reply:
I think that it would be better to remove it so that the code is simpler and has less going on. If an ability needs the optional in the future, the code can be changed then.

Reply:
Sure, I'll change that now